stackrox/acs-fleet-manager
# ROX-21679: add rotate secret backup feature to admin API

## Description

When secrets are rotated on data plane cluster side (e.g. leaf certs of central) they are not automatically updated in the backup of fleet-manager. This PR adds an admin API endpoint to trigger that rotation. Once we implement better working rotation mechanisms we should also address automating the backup rotation.

## Checklist (Definition of Done)


## Test manual
- Local fleet-manager & fleetshard-sync setup with `infractl` cluster
- Create central with `create-central.sh`
- Wait for full creation flow
- Call the new API
```
rhoas login --auth-url=https://auth.redhat.com/auth/realms/EmployeeIDP 
export OCM_TOKEN=$(rhoas authtoken)
./scripts/fmcurl "rhacs/v1/admin/centrals/$central_id/rotate-secrets" -XPOST -d '{"reset_secret_backup": true}'

make db/psql
Verify that secrets are deleted, and backup get restored after some time when FS reconciles.
```

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```
